### PR TITLE
[im3d] New port

### PR DESCRIPTION
**Pull request by RT2Code**
- #### What does your PR fix?
Add a new port : https://github.com/john-chapman/im3d

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
All, Yes

- #### Does your PR follow the [maintainer guide]
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/im3d/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        DESTINATION include
+    )
+endif()
+
+install(
+    EXPORT ${PROJECT_NAME}_target
+    NAMESPACE ${PROJECT_NAME}::
+    FILE ${PROJECT_NAME}-config.cmake
+    DESTINATION share/${PROJECT_NAME}
+)
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:45]: There should be a line break at the end of the file.

File: ports/im3d/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            ${CMAKE_CURRENT_SOURCE_DIR}/im3d_math.h
+        DESTINATION include
+    )
+endif()
+
+install(
+    EXPORT ${PROJECT_NAME}_target
+    NAMESPACE ${PROJECT_NAME}::
+    FILE ${PROJECT_NAME}-config.cmake
+    DESTINATION share/${PROJECT_NAME}
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:44]: This is not from upstream, so the config config should be named `unofficial-<pkg>` and use the `unofficial::<pkg>::` namespace.

File: ports/im3d/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.8)
+project(im3d)
+
+add_library(${PROJECT_NAME} "")
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:5]: This alias isn't needed.

File: ports/im3d/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.8)
+project(im3d)
+
+add_library(${PROJECT_NAME} "")
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+target_compile_options(${PROJECT_NAME} PRIVATE "-std=c++11")
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:7]: The C++ standard should be set using CMake properties.

File: ports/im3d/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	${PROJECT_NAME}
+	PUBLIC
+	   	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+		$<INSTALL_INTERFACE:include>
+)
+
+target_sources(
+    ${PROJECT_NAME}
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/im3d.cpp
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:19]: We do everything to deal with spaces in file paths, so we shouldn't break it in portfiles: Quote filepaths and similar expressions. And probably you can just drop `${CMAKE_CURRENT_SOURCE_DIR}` here.

File: ports/im3d/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    ARCHIVE DESTINATION lib
+    RUNTIME DESTINATION bin
+)
+
+if (NOT IM3D_SKIP_HEADERS)
+    install(
+        FILES
+            ${CMAKE_CURRENT_SOURCE_DIR}/im3d.h
+            ${CMAKE_CURRENT_SOURCE_DIR}/im3d_config.h
+            ${CMAKE_CURRENT_SOURCE_DIR}/im3d_math.h
```
Comment (dg0yt) [inline ports/im3d/CMakeLists.txt:35]: Quotes, or no `${CMAKE_CURRENT_SOURCE_DIR}`.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/im3d/CMakeLists.txt (hunk 2 above)
Comment (RT222) [reply ports/im3d/CMakeLists.txt]: Thanks, I was not aware of this new rule.

**Pull request merged**